src/activities/Elsa.Activities.Rpa.Web/Extensions/SeleniumExtensions.cs

```diff
@@ -0,0 +1,355 @@
+        }
+
+        public static TResult EnsureValue<TResult>(this IWebElement element, Func<IWebElement, TResult> function)
+        {
+            try
+            {
+                return function(element);
+            }
+            catch
+            {
+                return default(TResult);
+            }
+        }
+
+        public static TResult EnsureOperation<TResult>(this IWebDriver driver, Func<IWebDriver, TResult> function)
+        {
+            try
+            {
+                return function(driver);
+            }
+            catch
+            {
+                return default(TResult);
+            }
+        }
+
+        public static void ScrollToEnd(this IWebElement element)
+        {
+            var remote = (RemoteWebElement)element;
+            remote.WrappedDriver.ExecuteJavaScript("arguments[0].scrollLeft = arguments[0].offsetWidth", element);
+        }
+
+        public static void Scroll(this IWebElement element, Direction direction, int offset)
+        {
+            var remote = (RemoteWebElement)element;
+            var scroll = direction == Direction.Right || direction == Direction.Down
+                ? "+="
+                : direction == Direction.Left || direction == Direction.Up
+                    ? "-="
+                    : throw new Exception("Direction not supported");
+
+            var scrolling = direction == Direction.Right || direction == Direction.Left ? "scrollLeft" : "scrollTop";
+            remote.WrappedDriver.ExecuteJavaScript($"arguments[0].{scrolling} {scroll} {offset}", element);
+        }
+
+        public static void ScrollWindow(this IWebDriver driver, Direction direction, int offset)
+        {
+            var horizontalScroll = direction == Direction.Right
+              ? offset
+              : direction == Direction.Left
+                ? offset * -1
+                : 0;
+            var verticalScroll = direction == Direction.Down
+              ? offset
+              : direction == Direction.Up
+                ? offset * -1
+                : 0;
+            driver.ExecuteJavaScript($"window.scrollBy({horizontalScroll}, {verticalScroll});");
+        }
+
+        public static T GetProperty<T>(this IWebElement element, string name)
+        {
+            var remote = (RemoteWebElement)element;
+            return remote.WrappedDriver.ExecuteJavaScript<T>($"return arguments[0].{name}", element);
+        }
+        public static IQueryable<HtmlNode> GetHtmlNodes(this ISearchContext element, TimeSpan timeout = default)
+        {
+            if (timeout == default)
+                timeout = TimeSpan.FromSeconds(20);
+            var doc = new HtmlDocument();
+            var start = DateTime.Now;
+
+            IQueryable<HtmlNode> nodes = default;
+            string html = "";
+            while (string.IsNullOrEmpty(html) && (DateTime.Now - start) < timeout)
+            {
+                Task.Delay(100).GetAwaiter().GetResult();
```

Comment:
Any reason not to turn this entire method into an `async` method and do:

```csharp
await Task.Delay(100);
```

?